public/app/plugins/datasource/tempo/QueryEditor/QueryField.tsx

```diff
@@ -35,7 +35,7 @@ class TempoQueryFieldComponent extends React.PureComponent<Props> {
 
   async componentDidMount() {
     // Set initial query type to ensure traceID field appears
-    if (!this.props.query.queryType) {
+    if (!this.props.query.queryType || this.props.query.queryType === 'clear') {
```

Comment:
I am not sure I see any issue on main right now, if I refresh the explore page I still see the TraceID tab. Also What is `clear` type used for? It does not seem to be used for selecting a tab and I don't see it actually being set in the query anywhere. @joey-grafana maybe you know what it represents?

Reply:
I'm not sure why the `'clear'` string is used in particular but whenever you select a different query type, `queryType` is set to `'clear'`.

To reproduce the issue:
- Select the Tempo data source in Explore.
- Select a different query type (instead of staying on the default which is TraceID)
- Refresh the page and you will see that no query type is selected.

The PR changes this so that when you refresh the page, the default query type (TraceID) will be selected instead of no query type being selected.

Reply:
It's used to clear the results when swapping between tab i.e. search -> traceID -> Json upload and sets the [query type](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/tempo/datasource.ts#L56) to clear in doing so.

The idea here was that it would be nice if the query type was not set to clear when you refreshed the page so that at least one tab was selected. This is an edge case though and as discussed with Hamas, prob more of a UX item to chat about at standup rather than actually make a change.

Reply:
Ah ok yeah because then what is saved in the URL is the last executed query and that was the `clear` query and unless you run another query that does not change. Ok probably makes sense as a sort of quick fix for now. I mean it's a bit weird that you don't get the tab you have when you reload but that is probably for a bigger discussion.

Reply:
@CrypticSignal can you just add to the comment above this explaining a bit why the `this.props.query.queryType === 'clear'` is there.

Reply:
@aocenas Added in commit https://github.com/grafana/grafana/pull/53887/commits/4b912995257e8239cf39b546da102004b43ecf64.